tests/functional/u/unnecessary/unnecessary_dict_index_lookup.py

```diff
@@ -0,0 +1,63 @@
+# pylint: disable=missing-docstring, too-few-public-methods, expression-not-assigned, line-too-long
+
+a_dict = dict()
+b_dict = dict()
+
+for k,v in a_dict.items():
+    print(a_dict[k])  # [unnecessary-dict-index-lookup]
+    print(b_dict[k])  # Should not emit warning, accessing other dictionary
+    a_dict[k] = 123  # Should not emit warning, key access necessary
+    a_dict[k] += 123  # Should not emit warning, key access necessary
+    print(a_dict[k])  # [unnecessary-dict-index-lookup]
+    k = "another key"
+    print(a_dict[k])  # This is fine, key reassigned
+
+
+# Tests on comprehensions
+{v:1 for k,v in a_dict.items() if a_dict[k]}  # [unnecessary-dict-index-lookup]
+{v:1 for k,v in a_dict.items() if k}  # This is fine, no indexing
+{a_dict[k]:1 for k,v in a_dict.items() if k}  # [unnecessary-dict-index-lookup]
```

Comment:
```suggestion
{a_dict[k]: 1 for k, v in a_dict.items() if k}  # [unnecessary-dict-index-lookup]
```
This might be just me, but I would appreciate if you could add some whitespaces after `:` and `,`. There are a few instances of that here.

Reply:
maybe we should have a style checker for that

Reply:
fixed 0b414d3a

Reply:
We are not applying black on the functional tests automatically with pre-commit (because some functional tests need code that black would change). Maybe we could but this feel like heavy triage work. See https://github.com/PyCQA/pylint/blob/master/.pre-commit-config.yaml#L16

Reply:
I use flake8 with VS Code which automatically highlights it. In theory we could enable it for the tests as well, but as @Pierre-Sassoulas said this would be a lot of work.

IMO it isn't necessary to follow the style completely, at least for the tests. Just the whitespaces are something which can be quite annoying.